File: collect_app/src/test/java/org/odk/collect/android/support/FakeLifecycleOwner.java
package org.odk.collect.android.support;

import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LifecycleRegistry;

public class FakeLifecycleOwner implements LifecycleOwner {

    private final LifecycleRegistry lifecycle = new LifecycleRegistry(this);

    public FakeLifecycleOwner() {
        lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
    }

    public void destroy() {
        lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY);
    }

    @NonNull
    @Override
    public Lifecycle getLifecycle() {
        return lifecycle;
    }
}

File: android-test/src/main/java/org/odk/collect/androidtest/FakeLifecycleOwner.kt
package org.odk.collect.androidtest

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LifecycleRegistry

class FakeLifecycleOwner : LifecycleOwner {

    private val lifecycle: LifecycleRegistry by lazy {
        LifecycleRegistry(this).also {
            it.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
        }
    }

    fun destroy() {
        lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    }

    override fun getLifecycle(): Lifecycle {
        return lifecycle
    }
}
